www/app/ga-tracking.test.js

```diff
@@ -0,0 +1,75 @@
+import ReactGA from 'react-ga';
+
+import registerGoogleTracking from './ga-tracking';
+
+jest.mock('react-ga', () => ({
+    initialize: jest.fn(),
+    pageview: jest.fn(),
+}));
+
+const mockedRouter = {
+    asPath: '/foo',
+    events: {
+        on: jest.fn(),
+        off: jest.fn(),
+    },
+};
+
+describe('Google Analytics', () => {
+    beforeAll(() => {
+        process.env.GA_TRACKING_ID = '12345';
+    });
+
+    beforeEach(() => {
+        ReactGA.initialize.mockReset();
+        ReactGA.pageview.mockReset();
+        mockedRouter.events.on.mockReset();
+        mockedRouter.events.off.mockReset();
+    });
+
+    it('should attach tracking events to router', () => {
+        registerGoogleTracking(mockedRouter);
+
+        expect(mockedRouter.events.on).toHaveBeenCalledTimes(1);
+        expect(mockedRouter.events.on).toHaveBeenCalledWith('routeChangeComplete', expect.any(Function));
+    });
+
+    it('should register a page view if the route changes', () => {
+        let on;
+        const router = {
+            ...mockedRouter,
+            events: {
+                ...mockedRouter.events,
+                on: jest.fn((event, handler) => {
```

Comment:
We can simplify this. I would leave the mock implementation empty and do a check below, where we both assert that mockedRouter.events.on where called once and with the right args, and then I would grab the handler from .mock.calls[0][1].